portfolio/src/main/webapp/prototype.js

```diff
@@ -80,10 +80,27 @@ async function submitObject() {
     // If the load succeeded, display the analysis button.
     if (successful) {
         //ADD SOME PREVIEW HERE
-        document.getElementById(elements.analyzeButton).style.display = "block";
+        let request = new Request("/data?videoId=" + objectId, {method: "GET"});
+        let response = await fetch(request);
+        if (response.status >= 400) {
+            alert(await response.text());
+            return false;
+        }
+        // Update and show the preview with the response fields.
+        video = await response.json();
+        displayPreview(video)
+        document.getElementById(elements.analyzeButton).removeAttribute("style");
     }
 }
 
+    
+// Take a Video object and preview it
+function displayPreview(video) {
+    document.getElementById(elements.thumbnailImage).src = video.thumbnailUrl;
+    document.getElementById(elements.videoTitle).innerHTML = video.title;
+    document.getElementById(elements.previewSection).removeAttribute("display");
```

Comment:
You could resize the image so that it's the same size regardless of the thumbnail size. I can do this though if you want.